### PR TITLE
neuron: Add LLNL/Caliper option in the Spack build system.

### PR DESCRIPTION
**Pull request by ikitayama**
@pramodk could you review? This is for the `develop` branch.

### CODE REVIEW

File: var/spack/repos/builtin/packages/neuron/package.py [hunk 1]
```diff
@@ -32,6 +32,7 @@ class Neuron(CMakePackage):
     variant("python",        default=True,  description="Enable python")
     variant("rx3d",          default=False,  description="Enable cython translated 3-d rxd")
     variant("tests",         default=False, description="Enable unit tests")
+    variant("caliper",         default=False, description="Add LLNL/Caliper support")
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/neuron/package.py]: @ikitayama : LGTM,  just few nitpicks  - could you indent `default` with the line on `34`, `when` clause of line `47` with `46`? Then its good to go!

Thank you!

Comment (ikitayama) [reply var/spack/repos/builtin/packages/neuron/package.py]: Done.

**Review by pramodk — DISMISSED** [review]
LGTM

File: var/spack/repos/builtin/packages/neuron/package.py [hunk 2]
```diff
@@ -80,6 +82,9 @@ def cmake_options(spec_options):
         if "+legacy-unit" in spec:
             args.append('-DNRN_DYNAMIC_UNITS_USE_LEGACY=ON')
 
+        if "+caliper" in spec:
+            args.append('-DCORENRN_CALIPER_PROFILEING=ON')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/neuron/package.py]: Is this a typo:
```suggestion
            args.append('-DCORENRN_CALIPER_PROFILING=ON')
```
?

**Review by alalazo — CHANGES_REQUESTED** [review]
I think there's a typo, LGTM otherwise

File: var/spack/repos/builtin/packages/neuron/package.py (hunk 2 above)
Comment (pramodk) [reply var/spack/repos/builtin/packages/neuron/package.py]: of course it is! thank you @alalazo !

Comment (ikitayama) [reply var/spack/repos/builtin/packages/neuron/package.py]: Done.

**Review by alalazo — APPROVED** [review]

**Pull request merged**